applications/system-service/systemapi.h

```diff
@@ -211,6 +210,65 @@ class SystemAPI : public APIBase {
         toggleSwipeEnabled((SwipeDirection)direction);
     }
     void toggleSwipeEnabled(SwipeDirection direction){ setSwipeEnabled(direction, !getSwipeEnabled(direction)); }
+    Q_INVOKABLE void setSwipeLength(int direction, int length){
+        int maxLength;
+        if(!hasPermission("system")){
+            return;
+        }
+        if(direction <= SwipeDirection::None || direction > SwipeDirection::Down){
+            qDebug() << "Invalid swipe direction: " << direction;
+            return;
+        }
+        if (direction == SwipeDirection::Up || direction == SwipeDirection::Down){
+            maxLength = 1872;
+        }
+        else{
+            maxLength = 1404;
+        }
+        if (length < 0 || length > maxLength){
+            qDebug() << "Invalid swipe length: " << direction;
+            return;
+        }
+        setSwipeLength((SwipeDirection)direction, length);
+    }
+    Q_INVOKABLE void setAllSwipeLengths(int length){
```

Comment:
This is missing a permission check
```suggestion
    Q_INVOKABLE void setAllSwipeLengths(int length){
        if(!hasPermission("system")){
            return;
        }
```